teos/src/responder.rs

```diff
@@ -336,179 +309,60 @@ impl Responder {
-        &self,
-        txs: HashMap<UUID, (Transaction, Option<Transaction>)>,
-    ) -> (HashMap<UUID, ConfirmationStatus>, HashSet<UUID>) {
-        let mut accepted = HashMap::new();
-        let mut rejected = HashSet::new();
-
-        let mut trackers = self.trackers.lock().unwrap();
-        let mut carrier = self.carrier.lock().unwrap();
+    /// Returns a vector of rejected trackers.
+    fn rebroadcast_stale_txs(&self, height: u32) -> Vec<UUID> {
+        let dbm = self.dbm.lock().unwrap();
         let tx_index = self.tx_index.lock().unwrap();
+        let mut carrier = self.carrier.lock().unwrap();
+        let mut rejected = Vec::new();
 
-        for (uuid, (penalty_tx, dispute_tx)) in txs.into_iter() {
-            let status = if let Some(dispute_tx) = dispute_tx {
-                // The tracker was reorged out, and the dispute may potentially not be in the chain (or mempool) anymore.
-                if tx_index.contains_key(&dispute_tx.txid())
-                    | carrier.in_mempool(&dispute_tx.txid())
-                {
-                    // Dispute tx is on chain (or mempool), so we only need to care about the penalty
-                    carrier.send_transaction(&penalty_tx)
-                } else {
-                    // Dispute tx has also been reorged out, meaning that both transactions need to be broadcast.
-                    // DISCUSS: For lightning transactions, if the dispute has been reorged the penalty cannot make it to the network.
-                    // If we keep this general, the dispute can simply be a trigger and the penalty doesn't necessarily have to spend from it.
-                    // We'll keel it lightning specific, at least for now.
-                    let status = carrier.send_transaction(&dispute_tx);
-                    if let ConfirmationStatus::Rejected(e) = status {
-                        log::error!(
-                        "Reorged dispute transaction rejected during rebroadcast: {} (reason: {e})",
-                        dispute_tx.txid()
-                    );
-                        status
+        // TODO: Edit this db call to return only the in-mempool trackers so not to pull useless data.
+        for (uuid, tracker_summary) in dbm.load_tracker_summaries() {
+            if let ConfirmationStatus::InMempoolSince(h) = tracker_summary.status {
+                // Rebroadcast the penalty transaction if it has missed too many confirmations.
+                if (height - h) as u8 >= CONFIRMATIONS_BEFORE_RETRY {
+                    let tracker = dbm.load_tracker(uuid).unwrap();
+                    // If the dispute tx isn't on-chain, this means the breach has been reorged out.
+                    // Rebroadcast the dispute transaction.
+                    if !tx_index.contains_key(&tracker_summary.dispute_txid) {
+                        log::warn!(
+                            "Reorged dispute transaction {} for penalty transaction {} has missed many confirmations",
```

Comment:
Umm, I don't think this log makes much sense, given we are not tracking the dispute. What you want to see log here is that you detected a reorg of the dispute, so you're sending that out again

Reply:
Currently, the dispute is sent right away when a reorg happens. This one takes place when the reorged dispute isn't found in the new chain for a long time.